README.md

```diff
@@ -24,6 +24,57 @@ In order to limit cognitive overhead prefer:
 MVC + `actions` + `jobs` + `services` + `workers`
 
 - Actions receive & validate user input that create/update records, with any related business logic.
-- Job have a single purpose with a clear start and end, usually asynchronous and ideally idempotent. 
+- Job have a single purpose with a clear start and end, usually asynchronous and ideally idempotent.
 - Services encapsulate complex logic or external calls that can be used in multiple places, usually jobs and actions.
 - Workers are long running processes listening for events, usually triggering relevant job in response.
+
+
+## Front end structure, rules and guidelines
+To keep the code clean, consistent and well understood across all projects, we should follow the conventions, rules and guidelines below, in addition to the rules and conventions specified by the linting rules and generally those adopted by the language or framework/s.
+
+General conventions
+- directories should use camelCase multiword names
+- regular js modules should follow the same directory naming conventions for components, and should be named in camelCase
+- if the js module exports a class, the filename should be in PascalCase
+- avoid relative imports -- prefer absolute paths always. http://codeofrob.com/entries/stop-using-relative-paths-in-your-javascripts.html
+
+Component Naming Conventions:
```

Comment:
Maybe add here:
```
 - for single use components, where there will only be one on the app, prefix with `The`, e.g:
   - `TheHeader`
   - `TheMenu`
```

Reply:
Sounds good to me. Should probably add that a lot of a rules are following the official Vue style guide, in which this is one of them. Maybe I will add a link to those as well